### ISSUE TITLE
Peer review - vap43

### ISSUE THREAD

**Issue by vineetparikh** [TheoRepo/AutoMLproject#3]
This project proposal relates to AutoML, a burgeoning research field associated with automatically optimizing the hyperparameters for a machine learning model trained on a specific dataset (which could allow people who are unfamiliar with the intricacies of ML to create optimized systems for their own use cases quickly). More specifically, this project attempts to evaluate the performance of well-known AutoML algorithms (auto-sklearn, TPOT, etc.), find which algorithm works the best, and optimize its performance by reimplementing and finding more efficient methods of running this algorithm on a benchmark dataset. The dataset is unfortunately not specified.

One thing I really like about this project is its attempt to optimize the runtime of individual AutoML algorithms that have high accuracy but might be particularly slow: if AutoML is to be attainable and usable by a larger number of people who are analyzing increasingly large amounts of data, the algorithms that are used should be not only accurate, but efficient when dealing with data sources. However, this also presents an issue: it could be possible to overoptimize for individual model types or datasets, and there isn't much information on how this possible risk could be managed. Another cool part about this project is that it attempts to effectively compare different AutoML algorithms to find which one can be the most effective for any given usecase, which can be a useful tool for people who are interested in using AutoML to automate their workflow but have no idea as to where to start. Unfortunately, this presents another issue: the "benchmark dataset" that they mention isn't specifically stated and no metrics for explicit comparison are listed. Finally, this project is quite considerate of the larger issues at hand: there are more and more data sources that can be understood by domain-specific experts and can be analyzed effectively with ML, but domain-specific experts might not necessarily have the expertise to deal with optimizing an ML model for large datasets. However, this project doesn't necessarily articulate its own true contribution to the field of AutoML research or consider previous literature in the field (such as Efficient and robust automated machine learning by Feurer et al, a paper published in NIPS 2015 that explains auto-sklearn in more detail).
